### ISSUE TITLE
Style/TrailingCommaInArrayLiteral: inconsistent results when there are comments

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#14070]
## Expected behavior

```ruby
h1 = {
  a: 1,
  b: 2,
}
h2 = {
  a: 1,
  b: 2, # Some comment.
}

p h1, h2
```

IMHO, the `diff_comma` should not complain about the trailing comma in h2.  The documentation does state "only when that item is followed by an immediate newline.", but it looks much more like a documented bug than a genuine feature.

## Actual behavior

```
$ bundle exec rubocop rubocop/Style_TrailingCommaInArrayLiteral.rb
Inspecting 1 file
C

Offenses:

rubocop/Style_TrailingCommaInArrayLiteral.rb:1:1: C: Naming/FileName: The name of this source file (Style_TrailingCommaInArrayLiteral.rb) should use snake_case.
rubocop/Style_TrailingCommaInArrayLiteral.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
h1 = {
^
rubocop/Style_TrailingCommaInArrayLiteral.rb:7:7: C: [Correctable] Style/TrailingCommaInHashLiteral: Avoid comma after the last item of a hash, unless that item immediately precedes a newline.
  b: 2, # Some comment.
      ^

1 file inspected, 3 offenses detected, 2 offenses autocorrectable
```

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ bundle exec rubocop -V
1.75.2 (using Parser 3.3.7.4, Prism 1.4.0, rubocop-ast 1.44.0, analyzing as Ruby 3.3, running on ruby 3.3.7) [arm64-darwin24]
  - rubocop-minitest 0.38.0
  - rubocop-performance 1.25.0
```

Cheers!
